src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
+        this._initializeArrivalAirport(destination);
+        this._initializeArrivalRunway();
+        this._initializeFlightPlanAltitude(altitude, category, model);
+        this._initializePositionInRouteToBeginAtFixName(nextFix, category);
 
-        this.legCollection = this._buildLegCollection(route);
+        return this;
     }
 
     /**
-     * Destroy the instance and reset properties
+     * Reset class properties
      *
      * @for LegModel
-     * @method destroy
+     * @method reset
+     * @chainable
      */
-    destroy() {
-        this._navigationLibrary = null;
-        this._previousRouteSegments = [];
+    reset() {
+        this.arrivalAirportModel = null;
+        this.arrivalRunwayModel = null;
         this.currentPhase = '';
+        this.departureAirportModel = null;
         this.departureRunwayModel = null;
-        this.arrivalRunwayModel = null;
         this.flightPlanAltitude = INVALID_NUMBER;
-        this.legCollection = [];
-    }
+        this._routeModel = null;
+        this._navigationLibrary = null;
 
-    /**
-     * Return an array of waypoints in the flight plan that have altitude restrictions
-     *
-     * @for Fms
-     * @method getAltitudeRestrictedWaypoints
-     * @return {array<WaypointModel>}
-     */
-    getAltitudeRestrictedWaypoints() {
-        return this.waypoints.filter((waypoint) => waypoint.hasAltitudeRestriction);
+        return this;
     }
 
     /**
-     * Collects the `.getProcedureTopAltitude()` value from each `LegModel`
-     * in the `#legCollection`, then finds and returns the highest value
+     * Initialize #arrivalAirportModel
      *
-     * @for LegModel
-     * @method getTopAltitude
-     * @return {number}
+     * @for Fms
+     * @method _initializeArrivalAirport
+     * @param destinationIcao {string} ICAO identifier specified by spawn pattern
+     * @private
      */
-    getTopAltitude() {
-        const maxAltitudeFromLegs = _map(this.legCollection, (leg) => leg.getProcedureTopAltitude());
+    _initializeArrivalAirport(destinationIcao) {
+        if (destinationIcao === '') {
+            return;
+        }
 
-        return Math.max(...maxAltitudeFromLegs);
+        this.arrivalAirportModel = AirportController.airport_get(destinationIcao);
     }
 
     /**
-     * Collects the `.getProcedureBottomAltitude()` value from each `LegModel`
-     * in the `#legCollection`, then finds and returns the lowest value
+     * Initialize #arrivalRunwayModel
      *
-     * @for LegModel
-     * @method getBottomAltitude
-     * @return {number}
+     * @for Fms
+     * @method _initializeArrivalRunway
+     * @private
      */
-    getBottomAltitude() {
-        const valueToExclude = INVALID_NUMBER;
-        const minAltitudeFromLegs = _without(
-            _map(this.legCollection, (leg) => leg.getProcedureBottomAltitude()),
-            valueToExclude
-        );
+    _initializeArrivalRunway() {
+        if (this.arrivalAirportModel === null) {
```

Comment:
assuming this is initialized as `null`, a simple `!this.arrivalAirportModel` should work here.  Using `=== null` is not doing what you think and can lead to errors in the future.  

I'd prefer `!this.arrivalAirportModel` instead of  `this.arrivalAirportModel === null` in this case (and any others using `=== null`).